clients/js/src/AdminClient.ts

```diff
@@ -0,0 +1,217 @@
+import { Configuration, ApiApi as DefaultApi } from "./generated";
+import { handleSuccess, handleError } from "./utils";
+import { ConfigOptions } from './types';
+import {
+    AuthOptions,
+    ClientAuthProtocolAdapter,
+    IsomorphicFetchClientAuthProtocolAdapter
+} from "./auth";
+
+const DEFAULT_TENANT = "default_tenant"
+const DEFAULT_DATABASE = "default_database"
+
+// interface for tenant
+interface Tenant {
+    name: string,
+}
+
+// interface for tenant
+interface Database {
+    name: string,
+}
+
+export class AdminClient {
+    /**
+     * @ignore
+     */
+    private api: DefaultApi & ConfigOptions;
+    private apiAdapter: ClientAuthProtocolAdapter<any>|undefined;
+    private tenant: string = DEFAULT_TENANT;
+    private database: string = DEFAULT_DATABASE;
+
+    /**
+     * Creates a new AdminClient instance.
+     * @param {Object} params - The parameters for creating a new client
+     * @param {string} [params.path] - The base path for the Chroma API.
+     * @returns {AdminClient} A new AdminClient instance.
+     *
+     * @example
+     * ```typescript
+     * const client = new AdminClient({
+     *   path: "http://localhost:8000"
+     * });
+     * ```
+     */
+    constructor({
+        path,
+        fetchOptions,
+        auth,
+        tenant = DEFAULT_TENANT,
+        database = DEFAULT_DATABASE
+    }: {
+        path?: string,
+        fetchOptions?: RequestInit,
+        auth?: AuthOptions,
+        tenant?: string,
+        database?: string,
+    } = {}) {
+        if (path === undefined) path = "http://localhost:8000";
+        this.tenant = tenant;
+        this.database = database;
+
+        const apiConfig: Configuration = new Configuration({
+            basePath: path,
+        });
+        if (auth !== undefined) {
+            this.apiAdapter = new IsomorphicFetchClientAuthProtocolAdapter(new DefaultApi(apiConfig), auth);
+            this.api = this.apiAdapter.getApi();
+        } else {
+            this.api = new DefaultApi(apiConfig);
+        }
+
+        this.api.options = fetchOptions ?? {};
+    }
+
+    /**
+     * Creates a new tenant with the specified properties.
+     *
+     * @param {Object} params - The parameters for creating a new tenant.
+     * @param {string} params.name - The name of the tenant.
+     *
+     * @returns {Promise<Tenant>} A promise that resolves to the created tenant.
+     * @throws {Error} If there is an issue creating the tenant.
+     *
+     * @example
+     * ```typescript
+     * await adminClient.createTenant({
+     *   name: "my_tenant",
+     * });
+     * ```
+     */
+    public async createTenant({
+        name,
+    }: {
+        name: string,
+    }): Promise<Tenant> {
+        const newTenant = await this.api
+            .createTenant({name}, this.api.options)
+            .then(handleSuccess)
+            .catch(handleError);
+
+        // TODO: newTenant is null...... is this a bug in the backend API?
```

Comment:
<img width="591" alt="Screenshot 2023-11-15 at 9 12 45 PM" src="https://github.com/chroma-core/chroma/assets/5598697/bd905355-f8b5-4466-b395-98d1f9d31a8a">

Return type is None.

Reply:
is `None` the case of success? and `!None` a failure?

Reply:
So it'll always be none. Agreed it's not the best api (in the other comment). For now I'd rely on the status code to tell you of success or not

Reply:
sounds good